src/algorithms/derivatives/hamiltonian_derivatives.jl

```diff
@@ -323,11 +334,9 @@ function (H::JordanMPO_AC2_Hamiltonian)(x::MPOTensor)
     ismissing(H.DE) || @plansor y[-1 -2; -3 -4] += x[-1 1; -3 -4] * H.DE[-2; 1]
     ismissing(H.EE) || @plansor y[-1 -2; -3 -4] += x[1 -2; -3 -4] * H.EE[-1; 1]
 
-    GL, A1, A2, GR = H.AA
-    if nonzero_length(A1) > 0 && nonzero_length(A2) > 0
-        # TODO: there are too many entries here, this could be further optimized
-        @plansor y[-1 -2; -3 -4] += GL[-1 7; 6] * x[6 5; 1 3] * A1[7 -2; 5 4] *
-            A2[4 -4; 3 2] * GR[1 2; -3]
+    if !ismissing(H.AA)
+        GLW, GWR = H.AA
+        @tensor y[-1 -2; -3 -4] += GLW[-1 -2 3; 1 2] * x[1 2; 4 5] * GWR[3 4 5; -3 -4]
```

Comment:
Same comment here, you want to construct GWR such that the contraction ends up as:

```suggestion
        @plansor y[-1 -2; -3 -4] += GLW[-1 -2 5; 1 2] * x[1 2; 3 4] * GWR[3 4 5; -3 -4]
```